/*  Copyright 2012 MaidSafe.net limited

    This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
    version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
    licence you accepted on initial access to the Software (the "Licences").

    By contributing code to the MaidSafe Software, or to this project generally, you agree to be
    bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
    directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
    available at: http://www.maidsafe.net/licenses

    Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
    under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
    OF ANY KIND, either express or implied.

    See the Licences for the specific language governing permissions and limitations relating to
    use of the MaidSafe Software.                                                                 */

option optimize_for = LITE_RUNTIME;

package maidsafe.vault.protobuf;

message PmidManagerValue {
  required int64 stored_total_size = 1;
  required int64 lost_total_size = 2;
  required int64 offered_space = 3;
}

message PmidManagerKeyValuePair {
  required bytes key = 1;
  required bytes value = 2;
}

message PmidAccountResponse {
  required PmidAccount pmid_account = 1;
  required int32 status = 2;
}

message PmidAccount {
  required bytes pmid_name = 1;
  required bytes serialised_account_details = 2;
}

message PmidAccountDetails {
  required bytes serialised_pmid_manager_metadata = 1;
  repeated bytes serialised_unresolved_entry = 2;
}

message PutResult {
  required bool result = 1;
  required bytes pmid_name = 2;
  optional int32 data_size = 3;
  optional bytes serialised_data = 4;
}

message StateChange {
  required bytes pmid_name = 1;
}
